consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -301,9 +284,12 @@ impl<T: Payload> PersistentStorage<T> for StorageWriteProxy {
         );
 
         // find the block corresponding to storage latest ledger info
-        let (_, ledger_info, _, _) = read_client
-            .update_to_latest_ledger(0, vec![])
-            .expect("unable to read ledger info from storage");
+        let ledger_info = self
+            .read_client
+            .get_startup_info()
+            .expect("unable to read ledger info from storage")
+            .expect("startup info is None")
+            .ledger_info;
         let mut initial_data = RecoveryData::new(
```

Comment:
nit: not introduced in this PR. We're using the `initial_data` name for two different things in this functions, which is confusing. Could you name one of them differently?